readiness-tool/popup.js

```diff
@@ -0,0 +1,218 @@
+ * @param {Objected} detectedApps - All 3rd Party Applications found on page
+ */
+function showSupportedAppsInView(detectedApps) {
+  supportedAds.innerHTML = supportedAnalytics.innerHTML = notSupportedAds.innerHTML = notSupportedAnalytics.innerHTML = blankMessage;
+  supportedAds.appendChild(makeList(detectedApps.supported.ads, false));
+  supportedAnalytics.appendChild(makeList(detectedApps.supported.analytics, false));
+  notSupportedAds.appendChild(makeList(detectedApps.notSupported.ads, true));
+  notSupportedAnalytics.appendChild(makeList(detectedApps.notSupported.analytics, true));
+}
+/**
+ * Splits all detected apps into 'supported' and 'not supported'
+ * @param {String} htmlString - String containing all HTML on the page
+ * @return {Object} 
+ */
+function filterApps(htmlString) {
+  const foundThis = {
+    'supported': {
+      'ads': [],
+      'analytics': []
+    },
+    'notSupported': {
+      'ads': [], 
+      'analytics': []
+    }
+  };
+  // for all the app objects in the apps.JSON file
+  Object.keys(listAllApps).forEach(function (key) {
+    let val = listAllApps[key];
+    // If object has a 'script' key
+    if (val.script != null) {
+      Object.keys(val.script).forEach(function (x) {
+        addToDict(val.script[x].split('\\;')[0], htmlString, foundThis, key, val.cats);
+      });
+    }
+  });
+  return foundThis;
+}
+/**
+ * Pushes keys to the supported or not supported list of the object 'found this'
+ * @param {String} tempScript - String representation of regular expression
+ * @param {!String} htmlString - String containing all HTML on the page
+ * @param {Object} foundThis - Object separating the 3P services by support
+ * @param {String} key - name of third party service
+ * @param {String} category - the category that the key belongs to
+ */
+function addToDict(tempScript, htmlString, foundThis, key, category) {
+  const regX = new RegExp(tempScript);
+  if (regX.test(htmlString)) {
+    if (isKeyUnique(foundThis, key)) {
+      switch (true) {
+      case (isSupported(key) == true) && category == 10:
+        foundThis.supported.analytics.push(key);
+        break;
+      case (isSupported(key) == true) && category == 36:
+        foundThis.supported.ads.push(key);
+        break;
+      case (isSupported(key) == false) && category == 10:
+        foundThis.notSupported.analytics.push(key);
+        break;
+      case isSupported(key) == false && category == 36:
+        foundThis.notSupported.ads.push(key);
+        break;
+      }
+    }
+  }
+}
+/**
+ * Checks to see if key is unique within the object
+ * @param {Object} obj - Object separating the 3p services by support
+ * @param {String} key - name of third party service
+ * @return {boolean} 
+ */
+function isKeyUnique(obj, key) {
+  const truthValue = obj.supported.ads.includes(key) + 
+        obj.supported.analytics.includes(key) + 
+        obj.notSupported.ads.includes(key) + 
+        obj.notSupported.analytics.includes(key);
```

Comment:
`Array#includes` returns a boolean, so you're adding booleans together here. E.g.
```js
true + true + false + true
```